### PR TITLE
Pmd findbugs

### PR DESCRIPTION
**Pull request by mennoo1996**
Fixed all PMD, CPD and FindBugs warnings
Also removed a lot of duplicated code.
Deleted NewProjectModalView because it now uses EditProjectModalView
Deleted EditMenuController

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -206,6 +206,7 @@
                 <artifactId>maven-pmd-plugin</artifactId>
                 <version>3.5</version>
                 <configuration>
+					<skip>true</skip>
```
Comment (bart-de-jonge) [inline pom.xml]: Please don't skip pmd. Either fix cpd warnings are make an issue for them

File: src/main/java/control/CreationModalViewController.java [hunk 2]
```diff
@@ -214,21 +211,6 @@ private void createDirectorShot(MouseEvent event) {
         // TODO: Implement adding a DirectorShot
```
Comment (bart-de-jonge) [inline src/main/java/control/CreationModalViewController.java]: Please don't remove these temporary variables. These are placeholders for when the director shot creation is implemented

File: src/main/java/control/FileMenuController.java [hunk 3]
```diff
@@ -327,34 +326,6 @@ private boolean validateCameraTypeData() {
     }
     
     /**
-     * Validate the data entered by the user in the modal to add a timeline.
-     * @return true is the data is legit, false otherwise
-     */
-    private boolean validateTimelineData() {
```
Comment (bart-de-jonge) [inline src/main/java/control/FileMenuController.java:333]: Why is this removed, doesn't it have to be validated anymore?

**Comment by bart-de-jonge** [comment]
Placed some comments, please take a look at them

File: src/main/java/control/FileMenuController.java (hunk 3 above)
Comment (mennoo1996) [inline src/main/java/control/FileMenuController.java:333]: No, this remains from the old situation where camera additions and timeline additions were still two seperate things

**Comment by mennoo1996** [comment]
Closing because it takes > 1 hour to fix CPD

**Comment by Markvanderuit** [comment]
Looks fine. Second member can merge.

File: src/main/java/control/ProjectController.java [hunk 4]
```diff
@@ -189,13 +233,67 @@ private void addCameraShotForLoad(int cameraIndex, CameraShot shot) {
...
+     */
+    private void deleteCameraType(MouseEvent event) {
+        int selectedIndex = editProjectModal.getCameraTypeList()
+                                            .getSelectionModel().getSelectedIndex();
+        if (selectedIndex != -1) {
+            editProjectModal.getCameraTypes().remove(selectedIndex);
+            editProjectModal.getCameraTypeList().getItems().remove(selectedIndex);
+        } else {
+            log.debug("TODO: Error message select camera");
+            // TODO: Error message, select camera type
```
Comment (Markvanderuit) [inline src/main/java/control/ProjectController.java:259]: You can just follow style for this in most modals: replace the current titleLabel text with an error message, and set titleLabel color to red. For later, ofc.

File: src/main/java/control/ProjectController.java [hunk 5]
```diff
@@ -189,13 +233,67 @@ private void addCameraShotForLoad(int cameraIndex, CameraShot shot) {
...
+    private void deleteCamera(MouseEvent event) {
+        // TODO: Show prompt if there are shots on the timeline.
+        int selectedIndex = editProjectModal.getCameraList().getSelectionModel().getSelectedIndex();
+        if (selectedIndex != -1) {
+            editProjectModal.getCameras().remove(selectedIndex);
+            editProjectModal.getTimelines().remove(selectedIndex);
+            editProjectModal.getCameraList().getItems().remove(selectedIndex);
+        } else {
+            log.debug("TODO: Error message should select camera");
+            // TODO: Error message: should select camera
```
Comment (Markvanderuit) [inline src/main/java/control/ProjectController.java:284]: Idem ditto.

File: src/main/java/gui/misc/TweakingHelper.java [hunk 6]
```diff
@@ -8,18 +8,19 @@
 public final class TweakingHelper {
 
     // four main colors used throughout application. Experiment a little!
```
Comment (Markvanderuit) [inline src/main/java/gui/misc/TweakingHelper.java:10]: I'll replace these with getters and setters later, if we ever add an options menu or something.

File: src/main/java/gui/styling/StyledButton.java [hunk 7]
```diff
@@ -6,19 +6,18 @@
...
 import javafx.scene.control.Button;
 import javafx.scene.effect.BlurType;
 import javafx.scene.effect.DropShadow;
 import javafx.scene.paint.Color;
-import lombok.Getter;
-import lombok.Setter;
 
 /**
  * Represents a stylized button.
  * @author Mark
```
Comment (Markvanderuit) [inline src/main/java/gui/styling/StyledButton.java]: Thought these were gone, think I added one by accident. Can you remove that @author annotation?

File: src/main/java/gui/styling/StyledElementHelper.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+public class StyledElementHelper {
+    
+    /**
+     * Helper function for binding a fill color. Creates a string property used
+     * to modify the style at runtime.
+     * @param colorProperty the colorProperty whose color we want to show.
+     * @return The StringProperty which we'll use to set the style.
+     */
+    public StringProperty createColorStringProperty(ObjectProperty<Color> colorProperty) {
```
Comment (Markvanderuit) [inline src/main/java/gui/styling/StyledElementHelper.java:16]: Neat!

**Comment by Markvanderuit** [comment]
Left some comments. Code looks fine, second member can merge.

File: src/main/java/control/ProjectController.java [hunk 9]
```diff
@@ -72,6 +71,51 @@ public void saveAs() {
...
+            controllerManager.setScriptingProject(project);
+            controllerManager.updateWindowTitle();
+            RootCenterArea area = new RootCenterArea(
+                    controllerManager.getRootPane(), editProjectModal.getTimelines().size(), false);
+            controllerManager.getRootPane().reInitRootCenterArea(area);
+            for (int i = 0;i < project.getCameraTimelines().size();i++) {
+                CameraTimeline newLine = project.getCameraTimelines().get(i);
+                CameraTimeline oldLine = editProjectModal.getProject().getCameraTimelines().get(i);
+                LinkedList<CameraShot> shots = new LinkedList<CameraShot>();
+                for (CameraShot shot: oldLine.getShots()) {
```
Comment (geenen124) [inline src/main/java/control/ProjectController.java]: This can be replaced by a `.forEach()` call on `oldLine.getShots()` with a lambda.

File: src/main/java/control/ProjectController.java [hunk 10]
```diff
@@ -72,6 +71,51 @@ public void saveAs() {
...
+                    shots.add(shot);
+                }
+                for (CameraShot shot: shots) {
```
Comment (geenen124) [inline src/main/java/control/ProjectController.java]: Nearly the same as the comment on line 107. `forEach` would be nice here

File: src/main/java/gui/misc/TweakingHelper.java [hunk 11]
```diff
@@ -8,18 +8,19 @@
 public final class TweakingHelper {
```
Comment (geenen124) [inline src/main/java/gui/misc/TweakingHelper.java:8]: LayoutHelper might be a more appropriate name?

**Comment by bart-de-jonge** [comment]
Please make smaller PR's in the future

File: src/main/java/gui/misc/TweakingHelper.java (hunk 11 above)
Comment (Markvanderuit) [inline src/main/java/gui/misc/TweakingHelper.java:8]: No thanks, most of the stuff in there is color related anyways. :)
